python/cuml/_thirdparty/sklearn/utils/extmath.py

```diff
@@ -47,9 +42,13 @@ def row_norms(X, squared=False):
         The row-wise (squared) Euclidean norm of X.
     """
     if sparse.issparse(X):
-        if not isinstance(X, sparse.csr_matrix):
-            X = sparse.csr_matrix(X)
-        # norms = csr_row_norms(X)
+        if isinstance(X, (sparse.csr_matrix, sparse.csc_matrix,
+                          sparse.coo_matrix)):
+            X_copy = X.copy()
+            X_copy.data = np.square(X_copy.data)
```

Comment:
I'm guessing the answer to this is obvious, but you can't just do: ```
 X_sq = np.square(X)
 norms = X.sum(axis=1).squeeze()
```

Also, should consider updating the docstring which says that this won't create a temp copy. (here it will for sparse only)

Reply:
I updated the docstring. However, as strange as it may seem, it looks like CuPy can only square dense arrays as of now.